### PR TITLE
Mtdrewski

### PR DESCRIPTION
**Pull request by natanjel**

### CODE REVIEW

File: GRAPH-moment.iml [hunk 1]
```diff
@@ -23,5 +23,7 @@
     <orderEntry type="library" name="org.kordamp.ikonli:ikonli-materialdesign-pack:11.3.4" level="project" />
     <orderEntry type="library" name="org.kordamp.ikonli:ikonli-fontawesome-pack:11.3.4" level="project" />
     <orderEntry type="library" name="org.kordamp.ikonli:ikonli-javafx:11.3.4" level="project" />
+    <orderEntry type="library" name="ikonli-core-11.3.4" level="project" />
+    <orderEntry type="library" name="javafx-11" level="application" />
```
Comment (natanjel) [inline GRAPH-moment.iml:27]: This must have been added by IntelIJ automatically

File: src/com/github/mtdrewski/GRAPH_moment/model/dataProcessor/DataProcessor.java [hunk 2]
```diff
@@ -32,10 +32,17 @@ public Graph makeGraphFromInput(String textInput, Type tempType) throws Incorrec
...
         }
 
+        Random r = new Random();
+        float max = 700f;       //TODO: make it change with the Panel size
+        float min = 0f;
+        for (Vertex v1 : inputGraph.getVertices()) {
+            v1.setxPos(min + r.nextFloat() * (max - min));
+            v1.setyPos(min + r.nextFloat() * (max - min));
+        }
+
```
Comment (natanjel) [inline src/com/github/mtdrewski/GRAPH_moment/model/dataProcessor/DataProcessor.java:42]: For a while it creates nodes by randomly generating the coords

**Comment by natanjel** [comment]
Okay, it should be ready to merge

File: src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
 import javafx.scene.input.MouseButton;
 import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
 import java.util.function.Supplier;
 
 public class GraphDrawer {
```
Comment (mateuszpach) [inline src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java]: According to the new convention it should be GraphDrawerController

File: src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java [hunk 4]
```diff
@@ -93,4 +102,31 @@ protected void exitEdgeMode(boolean success) {
         sourceVertex = null;
         inEdgeMode = false;
     }
+
+    public void passGraphAndDraw(Graph graph1){
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java]: Consider renaming graph1 to something more semantic

File: src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+public class ImportController {
+
+    @FXML
+    TextArea textArea;
+
+    public GraphDrawer graphDrawer;
+
+    DataProcessor dataProcessor = new DataProcessor();
+
+    public void importGraphOnClick() throws IOException {
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java]: method name doesn't seem intuitive, consider renaming

File: src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            graph = dataProcessor.makeGraphFromInput(textInput, DataProcessor.Type.EDGE_LIST);
+        } catch (DataProcessor.IncorrectInputFormatException e) {
+            System.out.println(e.getMessage());         //TODO: maybe make this message as pop-up
+            return;
+        }
+        ((Stage) textArea.getScene().getWindow()).close();
+        System.out.println(graph);
+
+    }
+    public void canYouHearMe(){
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java]: If this is a debug function then remove it or at least mark as TODO

File: src/com/github/mtdrewski/GRAPH_moment/model/graphs/Vertex.java [hunk 7]
```diff
@@ -24,6 +24,14 @@ public int id() {
         return id;
     }
 
+    public void setxPos(float x){
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/model/graphs/Vertex.java:27]: Mayby a single function setPosition(x, y) would be better? (Shorter for sure)
As far as I know, we always set x and y in the same time

File: src/com/github/mtdrewski/GRAPH_moment/controller/NewProject.java [hunk 8]
```diff
@@ -1,4 +1,6 @@
 package com.github.mtdrewski.GRAPH_moment.controller;
 
 public class NewProject {
```
Comment (mateuszpach) [inline src/com/github/mtdrewski/GRAPH_moment/controller/NewProject.java]: According to the new convention it should be NewProjectController

File: src/com/github/mtdrewski/GRAPH_moment/Main.java [hunk 9]
```diff
@@ -7,15 +7,14 @@
...
 import javafx.stage.Stage;
 
-
 public class Main extends Application {
 
     @Override
     public void start(Stage primaryStage) throws Exception {
         FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("view/main.fxml"));
-        Parent root = fxmlLoader.load();
+        Parent root1 = fxmlLoader.load();
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/Main.java]: Why root changed to root1?

File: src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java [hunk 10]
```diff
@@ -115,14 +115,22 @@ private void reDrawGraph(){
             VertexCircle vertexShape= vertexShapeFactory.get();
             vertexShape.setCenterX(v1.xPos());
             vertexShape.setCenterY(v1.yPos());
-            anchorPane.getChildren().add(vertexShape);
             vertexCircles.add(vertexShape);
+            anchorPane.getChildren().add(vertexShape);
         }
 
         for(Edge e1:graph.getEdges()){
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java]: e1 is not a the best name

File: src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java [hunk 11]
```diff
@@ -115,14 +115,22 @@ private void reDrawGraph(){
...
-            EdgeLine edgeLine = edgeLineFactory.get();
-            edgeLine.setStartVertex(vertexCircles.get(e1.vert1().id()-1));
+
+            sourceVertex=vertexCircles.get(e1.vert1().id()-1);
```
Comment (woocash2) [inline src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java]: I didn't describe it anywhere so sorry for that but, sourceVertex is only to indicate a vertex from which there outgoes an edge which follows the mouse while being in "Edge Mode". So sourceVertex is only for purposes of Edge mode and you shouldn't use it for purpose of redrawing graph

File: src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java (hunk 6 above)
Comment (natanjel) [reply src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java]: Sorry, forgot to delete

Comment (natanjel) [reply src/com/github/mtdrewski/GRAPH_moment/controller/ImportController.java]: Wait, it is removen (outdated :p)

File: src/com/github/mtdrewski/GRAPH_moment/Main.java (hunk 9 above)
Comment (natanjel) [reply src/com/github/mtdrewski/GRAPH_moment/Main.java]: Outdated

Comment (natanjel) [reply src/com/github/mtdrewski/GRAPH_moment/Main.java]: Resolved in the latter commits

**Comment by mateuszpach** [comment]
Format code so it is consistent please. You can do it easily in IDE.

**Pull request merged**